scrapyd/website.py

```diff
@@ -53,178 +62,18 @@ def eggstorage(self):
-
-    def prep_doc(self):
-        return (
-            '<html>'
-            '<head>'
-            '<title>Scrapyd</title>'
-            '<style type="text/css">' + self.gen_css() + '</style>'
-            '</head>'
-            '<body><h1>Jobs</h1>'
-            '<p><a href="..">Go up</a></p>'
-            + self.prep_table() +
-            '</body>'
-            '</html>'
-        )
-
-    def prep_table(self):
-        return (
-            '<table id="jobs" border="1">'
-            '<thead>' + self.prep_row(self.header_cols) + '</thead>'
-            '<tbody>'
-            + '<tr><th colspan="%d">Pending</th></tr>' % len(self.header_cols)
-            + self.prep_tab_pending() +
-            '</tbody>'
-            '<tbody>'
-            + '<tr><th colspan="%d">Running</th></tr>' % len(self.header_cols)
-            + self.prep_tab_running() +
-            '</tbody>'
-            '<tbody>'
-            + '<tr><th colspan="%d">Finished</th></tr>' % len(self.header_cols)
-            + self.prep_tab_finished() +
-            '</tbody>'
-            '</table>'
-        )
-
-    def prep_tab_pending(self):
-        return '\n'.join(
-            self.prep_row(dict(
-                Project=project, Spider=m['name'], Job=m['_job'],
-                Cancel=self.cancel_button(project=project, jobid=m['_job'])
-            ))
-            for project, queue in self.root.poller.queues.items()
-            for m in queue.list()
-        )
-
-    def prep_tab_running(self):
-        return '\n'.join(
-            self.prep_row(dict(
-                Project=p.project, Spider=p.spider,
-                Job=p.job, PID=p.pid,
-                Start=microsec_trunc(p.start_time),
-                Runtime=microsec_trunc(datetime.now() - p.start_time),
-                Log='<a href="/logs/%s/%s/%s.log">Log</a>' % (p.project, p.spider, p.job),
-                Items='<a href="/items/%s/%s/%s.jl">Items</a>' % (p.project, p.spider, p.job),
-                Cancel=self.cancel_button(project=p.project, jobid=p.job)
-            ))
-            for p in self.root.launcher.processes.values()
-        )
-
-    def prep_tab_finished(self):
-        return '\n'.join(
-            self.prep_row(dict(
-                Project=p.project, Spider=p.spider,
-                Job=p.job,
-                Start=microsec_trunc(p.start_time),
-                Runtime=microsec_trunc(p.end_time - p.start_time),
-                Finish=microsec_trunc(p.end_time),
-                Log='<a href="/logs/%s/%s/%s.log">Log</a>' % (p.project, p.spider, p.job),
-                Items='<a href="/items/%s/%s/%s.jl">Items</a>' % (p.project, p.spider, p.job),
-            ))
-            for p in self.root.launcher.finished
-        )
-
-    def render(self, txrequest):
-        doc = self.prep_doc()
-        txrequest.setHeader('Content-Type', 'text/html; charset=utf-8')
-        doc = doc.encode('utf-8')
-        txrequest.setHeader('Content-Length', str(len(doc)))
```

Comment:
@mojtabaasadi setting content-type and content-length is probably fine to keep, or you see any reason to remove it?